includes/sanitizers/class-amp-tag-and-attribute-sanitizer.php

```diff
@@ -225,6 +225,7 @@ private function process_node( $node ) {
 
 		// Don't process text or comment nodes.
 		if ( XML_TEXT_NODE === $node->nodeType || XML_COMMENT_NODE === $node->nodeType || XML_CDATA_SECTION_NODE === $node->nodeType ) {
+			$this->set_plugin_output( $node );
```

Comment:
Let's refactor this to:

```php
if ( XML_COMMENT_NODE === $node->nodeType ) {
    $this->set_plugin_output( $node );
    return;
} elseif ( XML_TEXT_NODE === $node->nodeType || XML_CDATA_SECTION_NODE === $node->nodeType ) {
    return;
}
```

And then remove the check inside the `set_plugin_output` method. And that method can be renamed to something like `capture_current_source` as this should be for plugins and themes.

Reply:
Thanks, [these commits](https://github.com/Automattic/amp-wp/compare/fda14ea7f84fed6990d60571a629e013bb0053e4...3d1e6e3dd2a70ce4b7cb319f2bef0cfc56fc9856) use your code above to only send comment nodes to `set_plugin_output()`. They also rename that function to `capture_current_source()`, and remove a conditional inside it.